pkg/scheduler/framework/types.go

```diff
@@ -287,20 +290,22 @@ func newAffinityTerm(pod *v1.Pod, term *v1.PodAffinityTerm) (*AffinityTerm, erro
 	if err != nil {
 		return nil, err
 	}
+	at := &AffinityTerm{Namespaces: namespaces, Selector: selector, TopologyKey: term.TopologyKey, NamespaceSelector: nsSelector}
+	mergeAffinityTermMatchLabelKeys(at, term.MatchLabelKeys, pod.GetLabels(), enableMatchLabelKeysInPodAffinity)
```

Comment:
nit: how about do the featuregate check here, and thus save the parameter pass-in to mergeAffinityTermMatchLabelKeys:

```go
if enableMatchLabelKeysInPodAffinity {
    mergeAffinityTermMatchLabelKeys(at, term.MatchLabelKeys, pod.GetLabels())
    ...
```